backend/open_webui/utils/mcp_tools.py

```
@@ -318,42 +318,134 @@ async def _load_integration_tools(request: Request, user: UserModel) -> Dict[str
+                if user_token_data:
+                    refresh_result = None
+                    if user_token_data.refresh_token:
+                        try:
+                            refresh_result = refresh_microsoft_token(
+                                client_id,
+                                client_secret,
+                                tenant_id,
+                                user_token_data.refresh_token,
+                                SharePointMCPProvider.USER_SCOPES,
+                            )
+                        except Exception as exc:
```

Ruff (0.13.2)

Do not catch blind exception: `Exception`

(BLE001)